client/session.go

```diff
@@ -1429,17 +1526,12 @@ func (s *session) streamCollectedBlocksMetadata(
 		}
 
 		received, ok := metadata[m.id.Hash()]
-		if !ok {
+		if !ok || received.submitted {
 			received = &receivedBlocks{
 				results: make([]*blocksMetadata, 0, peersLen),
 			}
 			metadata[m.id.Hash()] = received
 		}
-		if received.submitted {
```

Comment:
Spoke to @xichen2020 offline. tl;dr - need the change because how we submit specify BlocksMetadata changes in Bootstrap v Repair. Next patch refactors this by making separate functions for the enqueue ops